syne_tune/optimizer/baselines.py

```diff
@@ -83,6 +84,18 @@ def __init__(self, config_space: Dict, metric: str, mode: str, **kwargs):
         )
 
 
+class REA(FIFOScheduler):
+    def __init__(self, config_space: Dict, metric: str, population_size: int = 100, sample_size: int = 10, **kwargs):
+        super(REA, self).__init__(
+            config_space=config_space,
+            metric=metric,
+            searcher=RegularizedEvolution(configspace=config_space, metric=metric,
```

Comment:
Please don't create the searcher by hand, this is not something we want to encourage people to do. Instead:

[...], searcher='kde', **kwargs)

The specific parameters can then be written into 'search_options' in kwargs.

Reply:
Can you elaborate why this is not something we want people to do? Otherwise it would cause additional friction to adapt the scheduler factory if people add a new searcher

Reply:
If you look into the constructor of FIFOScheduler, then if searcher is str, all sorts of required information is passed to the searcher_factory. All of this you have to remember to pass when creating your searcher on your own, and if you do not, something may break somewhere.

On top, if you go around searcher_factory, you may combine searchers with schedulers which do not go together. Again, this will not be checked then, and you get some very weird error somewhere.

Searcher is an internal concept. it is not en par with TrialScheduler, which is an external API.

And all other entries in baselines, except for Bore, go this safe way over searcher_factory.

Reply:
I don't understand why this creates friction. If I (for example) implement a new operator for PyTorch, I also need to link it into the structures where operators need to be linked in. Otherwise, things do not even work. So, the answer would be: if you write a new searcher, link it into searcher_factory.

Reply:
To explain this better: FIFOScheduler (or TrialScheduler) has certain arguments which need to also be passed to the searcher: metric (easy), points_to_evaluate (NOT easy, because of imputation), random_seed_generator / random_seed.

The searcher_factory plus mechanism in FIFOScheduler does all of that for you.

If you create BaseSearcher on your own, you have to do all of that.

Reply:
let's take this discussion offline